dsc_lib/src/discovery/command_discovery.rs

```diff
@@ -58,18 +58,27 @@ impl CommandDiscovery {
         };
 
         let mut paths = env::split_paths(&path_env).collect::<Vec<_>>();
+        // remove duplicate entries
+        let mut uniques = HashSet::new();
+        paths.retain(|e|uniques.insert((*e).clone()));
```

Comment:
shouldn't we remove dupes after any modifications in case PATH already includes dsc home?

Reply:
If it is already in the path then `if paths.contains(&exe_home_pb)` will be `true` and no modifications will be made.